src/SBarcodeDecoder.cpp

```diff
@@ -190,15 +199,70 @@ QImage SBarcodeDecoder::videoFrameToImage(QVideoFrame &videoFrame, const QRect &
         return image.rgbSwapped().copy(captureRect);
     }
 
+    #else
+
+    auto handleType = videoFrame.handleType();
+
+    qDebug() << handleType;
+
+    switch (handleType) {
+        case QVideoFrame::NoHandle: {
+
+            QImage image = videoFrame.toImage();
+
+            if (image.isNull()) {
+                return QImage();
+            }
+
+            if (image.format() != QImage::Format_ARGB32) {
+                image = image.convertToFormat(QImage::Format_ARGB32);
+            }
+
+            return image.copy(captureRect);
+
+        } break;
+        case QVideoFrame::RhiTextureHandle: { // NOP, videoFrame.handleType always returns NoHandle
+
+            QImage image(videoFrame.width(), videoFrame.height(), QImage::Format_ARGB32);
+
+            GLuint textureId = static_cast<GLuint>(1); // handle type enum 1 #videoFrame.handle().toInt()
+
+            QOpenGLContext *ctx = QOpenGLContext::currentContext();
+
+            QOpenGLFunctions *f = ctx->functions();
+
+            GLuint fbo;
+
+            f->glGenFramebuffers(1, &fbo);
+
+            GLint prevFbo;
+
+            f->glGetIntegerv(GL_FRAMEBUFFER_BINDING, &prevFbo);
+            f->glBindFramebuffer(GL_FRAMEBUFFER, fbo);
+            f->glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, textureId, 0);
+            f->glReadPixels(0, 0, videoFrame.width(), videoFrame.height(), GL_RGBA, GL_UNSIGNED_BYTE, image.bits());
+            f->glBindFramebuffer(GL_FRAMEBUFFER, static_cast<GLuint>( prevFbo ) );
+
+            return image.rgbSwapped().copy(captureRect);            
```

Comment:
Isn't This segment is basically the same with the version above? Would it be possible to make a smaller split?

Reply:
It has no operation cuz QVideoFrame returns always NoHandle type. It might deleted in the new commit.